src/nodes.coffee

```diff
@@ -573,7 +578,7 @@ exports.Value = class Value extends Base
         fst = new Value @base, @properties[...i]
         snd = new Value @base, @properties[i..]
         if fst.isComplex()
-          ref = new Literal o.scope.freeVariable 'ref'
+          ref = new IdentifierLiteral o.scope.freeVariable 'ref'
```

Comment:
There are 27 occurrences of `freeVariable` in nodes.coffee. I touched 11 of those in this commit: Those which had `new Literal` in front of them before. The rest use the return variable name as a string in different string concatenations.

It would also cause nodes.coffee and scope.litcoffee to become circular dependencies. While that would most likely still work, I think it's nice to avoid.

So, in my opinion, freeVariable would _ideally_ return an IdentifierLiteral, but right now it feels like too much work. Let's revisit that some other time.